internal/pkg/scaffold/olm-catalog/package_manifest.go

```diff
@@ -131,17 +134,25 @@ func (s *PackageManifest) newPackageManifest() *olmregistry.PackageManifest {
 // channel if possible.
 func (s *PackageManifest) setChannels(pm *olmregistry.PackageManifest) error {
 	if s.Channel != "" {
-		pm.Channels = append(pm.Channels, olmregistry.PackageChannel{
-			Name:           s.Channel,
-			CurrentCSVName: getCSVName(s.ProjectName, s.CSVVersion),
+		channelIdx := sort.Search(len(pm.Channels), func(i int) bool {
+			return pm.Channels[i].Name == s.Channel
```

Comment:
First time I'm seeing `sort.Search()` so I might be missing something, but how is binary search being used to find the desired channel name here?

```
// Search requires that
// f is false for some (possibly empty) prefix of the input range [0, n)
// and then true for the (possibly empty) remainder

For instance, given a slice data sorted in ascending order,
// the call Search(len(data), func(i int) bool { return data[i] >= 23 })
// returns the smallest index i such that data[i] >= 23. If the caller
// wants to find whether 23 is in the slice, it must test data[i] == 23
// separately.

// Searching data sorted in descending order would use the <=
// operator instead of the >= operator.
```

So isn't f supposed to be checking for `>=` or `<=` in a sorted ascending/descending array?

When `pm.Channels[i].Name == s.Channel` is `false` how does the function know which half of the channel array to discard to search for the correct name?

Reply:
Bleh this is what you get when you don't read docs properly. Changing to linear search.

Reply:
👍 Yeah moreover I think the channels aren't sorted by name are they? 
And even if they are I don't imagine they'll be big enough to warrant a binary search. 
Plus a simple linear search is easier to understand/maintain here too.

Reply:
Good catch @hasbro17. I breezed right over that.